velox/exec/Task.cpp

```diff
@@ -2433,6 +2434,12 @@ ContinueFuture Task::taskDeletionFuture() {
   return std::move(future);
 }
 
+std::string Task::printPlanWithStats(bool includeCustomStats) const {
+  VELOX_CHECK_NOT_NULL(planFragment_.planNode, "PlanFragment has no planNode");
```

Comment:
Would you add some tests that trigger this check and tests that invoke this API when task is in different states (before start, in progress, failed, succeeded)?

Reply:
@mbasmanova I added some tests to address this. Thanks.

Reply:
TaskTest.cpp also covers TaskStats update during Task cancel. This API only depends on planNode and TaskStats. We should have good test coverage now.